### PR TITLE
Update to NiFi release 1.16.1 and fixes

### PR DESCRIPTION
**Pull request by Chaffelson**
- modify utils.check_version to accept a default version to check against, and remove nested error when version check fails as it is overzealous
- Update test file paths in test_utils in case someone runs tests as root which actually can write to fake devices and therefore break the tests
- Added pylint exclusions for known but unimportant complaints
- Added workaround for NiFi 9470 by providing default inherited_parameter_contexts per advice from Chris Sampson
- Fixes https://github.com/Chaffelson/nipyapi/issues/305
- Updated NiFi and Registry versions in Docker compose configs to use latest 1.16.1 builds

### CODE REVIEW

File: nipyapi/utils.py [hunk 1]
```diff
@@ -470,7 +470,7 @@ def strip_snapshot(java_version):
     return re.sub("-SNAPSHOT", "", java_version)
```
Comment (ottobackwards) [inline nipyapi/utils.py:473]: should this be a constant? also the doc should say the default

**Comment by ottobackwards** [comment]
I have one comment about the default_version stuff.  Other than that this is super hard to review the way the commits are ( the middle commit has so many files ).

File: nipyapi/utils.py (hunk 1 above)
Comment (Chaffelson) [reply nipyapi/utils.py:473]: This was introduced in a previous version by you, so I have no idea 😹

Comment (ottobackwards) [reply nipyapi/utils.py:473]: Then it must be right!

**Comment by ottobackwards** [comment]
+1, obvious brilliance has gone into this

**Comment by Chaffelson** [comment]
Yeah fair, these PRs which update the baseline clients essentially replace the entire client subdirectory for nifi and registry as they are procedurally generated from the swagger. We have a goal in the future to perhaps build them on the fly as you suggested some time ago, but presently they are staticly generated.

**Review by ChrisSamo632 — APPROVED** [review]

**Comment by ChrisSamo632** [comment]
Other than the same comment about there being lots of files in the PR (maybe it would be better to do that as a separate branch/PR where just the swagger-generated files are updated until such as time as they're done automatically?)... I can see the few recent fixes here that were recently discussed

👍 from me

**Pull request merged**